### PR TITLE
Add `seed` to `TopKMetric` for consistent metrics when ties in ranking

### PR DESCRIPTION
**Pull request by oliverholworthy**
### Goals :soccer:

- Improve reliability of tests using top-k metrics 

### Implementation Details :construction:

- Adds `seed` parameter to TopKMetric classes to ensure results are consistent when evaluation run multiple times on the same data.
- Fixes `pre` parameter when using `model.predict` (typo in variable referenced)

### Testing Details :mag:

- Adds a test for `seed` parameter with NDCGAt

### CODE REVIEW

**Review by marcromeyn — APPROVED** [review]

File: merlin/models/tf/models/base.py [hunk 1]
```diff
@@ -696,7 +696,7 @@ def predict_step(self, data):
         x, _, _ = unpack_x_y_sample_weight(data)
 
         if getattr(self, "predict_pre", None):
-            out = call_layer(self.predict_pr, x, features=x, training=False)
+            out = call_layer(self.predict_pre, x, features=x, training=False)
```
Comment (oliverholworthy) [inline merlin/models/tf/models/base.py:699]: There was a typo here. we didn't have a test using model.predict with a `pre` .

**Comment by oliverholworthy** [comment]
rerun tests

**Comment by marcromeyn** [comment]
rerun tests

**Comment by marcromeyn** [comment]
rerun tests

**Comment by marcromeyn** [comment]
rerun tests

**Pull request merged**